src/VisualStudio/Core/Def/Implementation/CodeCleanup/CodeCleanupFixerProvider.cs

```diff
@@ -13,12 +13,9 @@
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.CodeCleanup
 {
-    /// <summary>
-    /// This is intentionally not exported as a concrete type and not an instance of
-    /// <see cref="ICodeCleanUpFixerProvider"/>. Roslyn is responsible for registering its own fixer provider, as
-    /// opposed to the implementation importing the instances of some interface.
-    /// </summary>
-    [Export]
+    [Export(typeof(ICodeCleanUpFixerProvider))]
+    [ExportMetadata("AppliesToProject", ContentTypeNames.CSharpContentType)]
```

Comment:
I have no idea. @olegtk do you know Chris' GitHub account to ask him?

Reply:
It's meant to be a new core MEF attribute (in MS.VS.CoreUtility.dll), similar to [CPS AppliesToAttribute](https://github.com/Microsoft/VSProjectSystem/blob/master/doc/extensibility/index.md) so MEF components could declare supported project system capabilities.
This way we can detect matching fixer providers for a given project/solution.

Reply:
But what does it mean to mix a content type and a project like this? And should there be a string constant for AppliesToProject rather than having to hardcode it?

Reply:
I see it this way: [ContentType] associates a component with a text buffer content type and [AppliesToProject] associates a component with a project capability. They are related but really orthogonal pieces of metadata. Depending on context you can mix both, for example ICodeCleanupFixerProvider can handle both text buffers and projects so it can have both attributes.

Good question about string constant. @AArnott, is there a list of predefined project capabilities somewhere?

Reply:
>  Can we just use AppliesToAttribute

First we need an answer to whether you can/should change the metadata name to match `AppliesTo`. If so, then technically you can use the `AppliesToAttribute` if you have or don't mind adding a reference to the CPS ref assembly.

Reply:
Ok, I will change the metadata to AppliesTo to match what CPS uses. `
namespace Microsoft.VisualStudio.Utilities
{
public sealed class AppliesToProjectAttribute : SingletonBaseMetadataAttribute
{
public string AppliesTo { get; }}
}
`